spec/system/provider_interface/change_existing_offer_ske_standard_flow_spec.rb

```diff
@@ -0,0 +1,271 @@
+    choose 'Yes'
+  end
+
+  def then_the_ske_reason_page_is_loaded
+    expect(page).to have_current_path("/provider/applications/#{application_choice.id}/offer/ske-reason/edit", ignore_query: true)
+  end
+
+  def when_i_add_a_ske_reason
+    choose t('provider_interface.offer.ske_reasons.form.different_degree', degree_subject: @ske_subject.name)
+  end
+
+  def then_the_ske_length_page_is_loaded
+    expect(page).to have_current_path("/provider/applications/#{application_choice.id}/offer/ske-length/edit", ignore_query: true)
+  end
+
+  def when_i_answer_the_ske_length
+    choose '8 weeks'
+  end
+
+  def and_the_ske_conditions_should_be_displayed
+    expect(page).to have_content('Subject knowledge enhancement course')
+    expect(page).to have_content("Subject\n#{@ske_subject.name}")
+    expect(page).to have_content("Length\n8 weeks")
+    expect(page).to have_content("Reason\nTheir degree subject was not #{@ske_subject.name}")
+  end
+
+  def then_the_conditions_page_is_loaded
+    expect(page).to have_content('Conditions of offer')
+  end
+
+  def and_the_conditions_of_the_original_offer_are_filled_in
+    expect(find("input[value='Fitness to train to teach check']")).to be_checked
+    expect(page).to have_field('Condition 1', with: 'Be cool')
+  end
+
+  def when_i_add_a_further_condition
+    click_on 'Add another condition'
+    fill_in('provider_interface_offer_wizard[further_conditions][1][text]', with: 'A* on Maths A Level')
+  end
+
+  def and_i_add_another_and_then_remove_a_further_condition
+    click_on 'Add another condition'
+    fill_in('provider_interface_offer_wizard[further_conditions][2][text]', with: 'Go to the cinema')
+    click_on 'Remove condition 3'
+  end
+
+  def then_the_correct_conditions_are_displayed
+    expect(page).to have_field('Condition 2', with: 'A* on Maths A Level')
+    expect(page).not_to have_field('Condition 3', with: 'Go to the cinema')
+  end
+
+  def then_the_review_page_is_loaded
+    expect(page).to have_content('Check and send new offer')
+  end
+
+  def and_i_can_confirm_the_changed_offer_details
+    within('.app-offer-panel') do
+      expect(page).to have_content(@selected_provider.name_and_code)
+      expect(page).to have_content(@selected_course.name_and_code)
+      expect(page).to have_content(@selected_course_option.study_mode.humanize)
+      expect(page).to have_content(@selected_course_option.site.name_and_address(' '))
+      expect(page).to have_content('Fitness to train to teach check')
+      expect(page).to have_content('Be cool')
+      expect(page).to have_content('A* on Maths A Level')
+    end
+  end
+
+  def when_i_send_the_offer
+    click_on 'Send new offer'
+  end
+
+  def then_i_see_that_the_offer_was_successfully_updated
+    within('.govuk-notification-banner--success') do
+      expect(page).to have_content('New offer sent')
+    end
+  end
+
+  def when_i_choose_to_change_the_conditions
+    click_on 'Add or change conditions'
+  end
+end
```

Comment:
Is it important to check on the DB we delete the old SKE conditions and created the new one?

Reply:
Yes, it makes sense. :)